### DISCUSSION TITLE
Lazygit: Open file with `e` in the current tab

### DISCUSSION THREAD

**Discussion by m4kamran** [folke/snacks.nvim#87] [General]
Currently when pressing e(edit) on a file in lazygit, it opens in a new tab. Is it possible to open it in the current tab instead?

**Comment by axkirillov** [comment]
Hi! Just add this to yout lazygit.yml
```yaml
os:
  edit: '[ -z "$NVIM" ] && (nvim -- {{filename}}) || (nvim --server "$NVIM" --remote-send "q" && nvim --server "$NVIM" --remote {{filename}})'
```

a config like this should also work, but for me it does not
```lua
return
{
  "folke/snacks.nvim",
  opts = {
    lazygit = {
      config = {
        os = {
          editPreset = "nvim-remote",
          --    this does not work, i dunno why
          edit =
          '[ -z "$NVIM" ] && (nvim -- {{filename}}) || (nvim --server "$NVIM" --remote-send "q" && nvim --server "$NVIM" --remote {{filename}})'
        }
      }
    }
  }
}
```

**Reply by m4kamran** [reply]
This works perfectly. Thanks! 🥳

**Reply by m4kamran** [reply]
@axkirillov Is it just me or does this not working anymore? using the latest nvim, lazygit and lazyvim. For some reason I even lose my cursor visibility after trying to edit.

**Reply by axkirillov** [reply]
@m4kamran I haven't updated yet. Which versions do you have?

**Reply by axkirillov** [reply]
@m4kamran Hi, it stopped working for me too, I dunno why
It works again once I remove the config in snacks.nvim and instead just do it in lazygit.yml
```yaml
os:
  edit: '[ -z "$NVIM" ] && (nvim -- {{filename}}) || (nvim --server "$NVIM" --remote-send "q" && nvim --server "$NVIM" --remote {{filename}})'
```

**Reply by aksh1618** [reply]
As mentioned in the README:

>  -- snacks does NOT have a full yaml parser, so if you need `"test"` to appear with the quotes
>  -- you need to double quote it: `"\"test\""`

Factoring this in, this works for me:

```lua
{
    "folke/snacks.nvim",
    opts = {
        lazygit = {
            config = {
                os = {
                    edit =
                        '[ -z "\"$NVIM\"" ] && (nvim -- {{filename}}) || (nvim --server "\"$NVIM\"" --remote-send "\"q\"" && nvim --server "\"$NVIM\"" --remote {{filename}})'
                }
            }
        },
    },
}
```

**Reply by fadarrizz** [reply]
I saw that this command is almost the same as the one implement in LazyGit as the `nvim-remote` preset, but with `--remote` instead of `--remote-tab`. Reference:
https://github.com/jesseduffield/lazygit/blob/master/pkg/config/editor_presets.go#L72-L80

The `nvim-remote` preset in LazyGit also includes `editAtLine` and `openDirInEditor`. I've tweaked these to open in the same tab instead of a new tab as well, resulting in:

```
  edit: '[ -z "$NVIM" ] && (nvim -- {{filename}}) || (nvim --server "$NVIM" --remote-send "q" && nvim --server "$NVIM" --remote {{filename}})'
  editAtLine: '[ -z "$NVIM" ] && (nvim +{{line}} -- {{filename}}) || (nvim --server "$NVIM" --remote-send "q" &&  nvim --server "$NVIM" --remote {{filename}} && nvim --server "$NVIM" --remote-send ":{{line}}<CR>")'
  openDirInEditor: '[ -z "$NVIM" ] && (nvim -- {{dir}}) || (nvim --server "$NVIM" --remote-send "q" && nvim --server "$NVIM" --remote {{dir}})'
  openInTerminal: '[ -z "$NVIM" ]'
```
